### PR TITLE
use fixed flat set for font farbling sets

### PR DESCRIPTION
**Pull request by arthuredelstein**
Resolves https://github.com/brave/brave-browser/issues/34094

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: third_party/blink/renderer/brave_font_whitelist.cc [hunk 1]
```diff
@@ -12,858 +12,828 @@ namespace brave {
 
 namespace {
 
-base::flat_set<std::string_view> kEmptyFontSet =
-    base::MakeFlatSet<std::string_view>(std::vector<std::string_view>{});
+bool simulate_empty_font_whitelist_for_testing = false;
```
Comment (goodov) [inline third_party/blink/renderer/brave_font_whitelist.cc:15]: `g_simulate_empty_font_whitelist_for_testing`

File: third_party/blink/renderer/brave_font_whitelist.cc [hunk 2]
```diff
@@ -12,858 +12,828 @@ namespace brave {
...
+    "small fonts",
+    "staccato222 bt",
+    "trajan pro",
+    "univers ce 55 medium",
+    "vrinda",
+    "zwadobef",
+});
 
 #if BUILDFLAG(IS_MAC)
 bool kCanRestrictFonts = true;
```
Comment (goodov) [inline third_party/blink/renderer/brave_font_whitelist.cc:75]: `constexpr` all `kCanRestrictFonts` vars.

File: third_party/blink/renderer/brave_font_whitelist.cc [hunk 3]
```diff
@@ -12,858 +12,828 @@ namespace brave {
...
+                                              "sans-serif-light",
+                                              "sans-serif-medium",
+                                              "sans-serif-monospace",
+                                              "sans-serif-thin",
+                                              "source sans pro",
+                                              "source-sans-pro-semi-bold",
+                                              "tahoma",
+                                              "times",
+                                              "times new roman",
+                                              "verdana"});
```
Comment (goodov) [inline third_party/blink/renderer/brave_font_whitelist.cc:729]: just do a mass replacement in this file: `"});` -> `",});` and run clang-format.

File: third_party/blink/renderer/brave_font_whitelist.cc
Comment (goodov) [inline third_party/blink/renderer/brave_font_whitelist.cc:1]: we may also rename whitelist -> allowlist everywhere here, but it's up to you.

File: third_party/blink/renderer/brave_font_whitelist.cc [hunk 4]
```diff
@@ -12,858 +12,828 @@ namespace brave {
...
+constexpr auto kFontWhitelistLO = kEmptyFontSet;
+constexpr auto kFontWhitelistML = kEmptyFontSet;
 #endif
 }  // namespace
 
 bool AllowFontByFamilyName(const AtomicString& family_name,
                            WTF::String default_language) {
   if (!kCanRestrictFonts)
     return true;
+  if (simulate_empty_font_whitelist_for_testing) {
```
Comment (goodov) [inline third_party/blink/renderer/brave_font_whitelist.cc:825]: `if (UNLIKELY(simulate_empty_font_whitelist_for_testing))`

**Review by goodov — COMMENTED** [review]
lgtm with few nits

File: third_party/blink/renderer/brave_font_whitelist.h [hunk 5]
```diff
@@ -23,18 +24,19 @@ BLINK_EXPORT bool IsFontAllowedForFarbling(const AtomicString& family_name);
...
+GetAdditionalFontWhitelistByLocale(WTF::String locale_language);
 
 // Testing-only functions
-BLINK_EXPORT void set_font_whitelist_for_testing(
-    bool can_restrict_fonts,
-    const base::flat_set<std::string_view>& font_whitelist);
 
-BLINK_EXPORT bool get_can_restrict_fonts_for_testing();
+// BLINK_EXPORT const base::span<std::string_view>&
+// GetFontWhitelistForTesting();
```
Comment (goodov) [inline third_party/blink/renderer/brave_font_whitelist.h:33]: del

File: third_party/blink/renderer/brave_font_whitelist.h [hunk 6]
```diff
@@ -9,7 +9,8 @@
 #include <string>
 #include <string_view>
 
-#include "base/containers/flat_set.h"
+#include "base/containers/fixed_flat_set.h"
+#include "base/no_destructor.h"
```
Comment (goodov) [inline third_party/blink/renderer/brave_font_whitelist.h:13]: these two headers shouldn't be necessary here.

File: browser/farbling/brave_font_whitelist_browsertest.cc [hunk 7]
```diff
@@ -67,12 +67,10 @@ class BraveFontWhitelistRenderViewTest : public content::RenderViewTest {
...
-  // Clear the font whitelist. This creates a situation where we know there is a
-  // font installed locally (Helvetica, preinstalled on every Mac) that is not
-  // on the font whitelist.
-  brave::set_font_whitelist_for_testing(
-      true,
-      base::MakeFlatSet<std::string_view>(std::vector<std::string_view>{}));
+  // Simulate an empty font whitelist. This creates a situation where we know
+  // there is a font installed locally (Helvetica, preinstalled on every Mac)
+  // that is not on the font whitelist.
+  brave::SetSimulateEmptyFontWhitelistForTesting(true);
```
Comment (goodov) [inline browser/farbling/brave_font_whitelist_browsertest.cc:73]: pls set it to false at the test.

File: third_party/blink/renderer/brave_font_whitelist.cc (hunk 1 above)
Comment (arthuredelstein) [reply third_party/blink/renderer/brave_font_whitelist.cc:15]: done

File: third_party/blink/renderer/brave_font_whitelist.cc (hunk 2 above)
Comment (arthuredelstein) [reply third_party/blink/renderer/brave_font_whitelist.cc:75]: done

File: third_party/blink/renderer/brave_font_whitelist.cc (hunk 3 above)
Comment (arthuredelstein) [reply third_party/blink/renderer/brave_font_whitelist.cc:729]: done

File: third_party/blink/renderer/brave_font_whitelist.cc
Comment (arthuredelstein) [reply third_party/blink/renderer/brave_font_whitelist.cc:1]: I will leave it for now, but feel free to open a ticket

File: third_party/blink/renderer/brave_font_whitelist.cc (hunk 4 above)
Comment (arthuredelstein) [reply third_party/blink/renderer/brave_font_whitelist.cc:825]: done

File: third_party/blink/renderer/brave_font_whitelist.h (hunk 5 above)
Comment (arthuredelstein) [reply third_party/blink/renderer/brave_font_whitelist.h:33]: done

File: third_party/blink/renderer/brave_font_whitelist.h (hunk 6 above)
Comment (arthuredelstein) [reply third_party/blink/renderer/brave_font_whitelist.h:13]: removed

File: browser/farbling/brave_font_whitelist_browsertest.cc (hunk 7 above)
Comment (arthuredelstein) [reply browser/farbling/brave_font_whitelist_browsertest.cc:73]: done

Comment (arthuredelstein) [reply browser/farbling/brave_font_whitelist_browsertest.cc:73]: done

**Comment by arthuredelstein** [comment]
Thanks for the review!

**Pull request merged**